src/analyses/base.ml

```diff
@@ -1145,7 +1155,12 @@ struct
          * side-effects here, but the code still distinguishes these cases. *)
       if (!GU.earlyglobs || ThreadFlag.is_multi a) && is_global a x then begin
         if M.tracing then M.tracel "setosek" ~var:x.vname "update_one_addr: update a global var '%s' ...\n" x.vname;
-        let new_value = update_offset (Priv.read_global a gs st x) in
+        let old_value = if Option.map_default exp_is_integral_var false lval_raw then
```

Comment:
It seems like it should be possible to apply this optimzation in the more general case that `Cil.isIntegralType lval_type` holds, but this somehow produces unreached fixed-points in multiple test cases, e.g. `02/24` that operates on malloced integers.

Reply:
The side effect is `(3)`, which I think is just an integer, but the solver has `((3), (4), true)`, which is the same integer, but inside a blob. There probably needs to be some special handling of blobs (or at least ignoring the optimization for them).

Reply:
Actually the zero init/origin boolean is confusingly flipped:
https://github.com/goblint/analyzer/blob/3d0cf6e1025814fa22a7cf1e57d578d5c9b3c4e5/src/cdomains/valueDomain.ml#L55-L61
If it's with `calloc`, then it's zero initialized, but then the boolean is `false`.

Reply:
>The side effect is (3), which I think is just an integer, but the solver has ((3), (4), true), which is the same integer, but inside a blob of size 4 (without (!) zero init, see next comment). There probably needs to be some special handling of blobs (or at least ignoring the optimization for them).

I now went with checking whether the lval is an allocated block and then not doing the optimization.